docs/enterprise/concepts.md

```diff
@@ -0,0 +1,140 @@
+
+### Hierarchical Policy Enforcement
+
+Hierarchical policy lets administrators enforce high level authorization policy. Policies can be optional (self-select), or mandatory.
+
+Identity and access is organized and controlled with Identity and Access Management (**IAM**) teams via your Identity Provider (**IdP**). These users and groups are read by Pomerium from your IdP, and used to determine top-level Namespace organization.
+
+Consider this scenario: your organization has a security team managing high-level, course grain authorization controls. For example, they want to ensure that everyone with access to internal resources:
+
+   - has a `yourcompany.com` email account,
+   - isn't coming from a known bad actor IP address,
+
+From there, the security team delegates management of child [Namespaces](#namespaces) to application teams, providing flexibility to self-manage their own application [Routes](#routes) and [Policies](#policies).
+
+For example, a developer group can be given control to determine who has access to their Namespace, and create or edit Routes within it. They can provide authentication to their WiP app without writing new authorization code. Meanwhile, the CFO is given [manager](#manager) permissions over the "Accounting" Namespace, and can set enforced or optional policies for the services within.
+
+### RBAC for Enterprise Console Users
+
+- Namespaces are also used to achieve Role Based Access Control (**RBAC**) in the console itself.
+- There are three different roles:
+
+#### Guest (no role)
+
+Users who are authenticated by your IdP but do not have a role assigned in the Pomerium Console can still view the list of Namespaces, but nothing else.
+
+#### Viewer
+
+A user with the Viewer role can:
+
+- view all resources in a Namespace (Routes, Policies, Certificates), including child Namespaces
+- view traffic dashboard for routes in the Namespace, including child Namespaces
+- view the activity log for a namespace.
+
+#### Manager
+
+In addition to the access provided by the Viewer role, the Manager can create, read, update, and delete routes, policies, and certificates in a Namespace (as wel as its children). A Manager may also reference policies and certificates in the parent Namespace.
+
+#### Admin
+
+An Admin user has permissions across all Namespaces. They can manage global settings, sessions, and service accounts, as well as view events and runtime data.
+
+## Users and Groups
+
+Pomerium populates users and groups from your IdP. This data is cached to prevent hitting API rate-limits, and provides look-ahead support when adding users or groups to [Namespaces](#namespaces) and [Policies](#policies).
+
+### Non-Domain Users
+
+When using Google as your IdP, you may hit an edge case where users are not available for lookup when the user's domain doesn't match that of your organization. In this case, there are two workarounds:
```

Comment:
```suggestion
You encounter a situation where you may want to add users that are not directly associated with your corporate identity provider service. For example, you have a corporate GSuite account and want to add a contractor with a gmail account. In this case, there are two workarounds:
```

Reply:
Not quite right, but I think this exists outside of just google (github, gitlab) where users may or may not be part of your organization but there is a shared directory, but is certainly most common in google.

Reply:
This may also be the case with Azure too; so best to just treat generically.

Reply:
edited to be _quite_ right, thanks!